src/main/scala/rps/controller/GameController.scala

```diff
@@ -1,40 +1,39 @@
-package rps.controller
+package rps.controller.GameController
+
+import scala.concurrent.{ExecutionContext, Future}
 
 import rps.model.{GameSummary, Move, Result}
 import Move._, Result._
 import rps.model.error._
 import rps.service.{GameService}
-import rps.repository.{GameRepository}
-import GameRepository._
-
-import scala.concurrent.{ExecutionContext, Future}
+import rps.repository.{InMemoryGameRepository}
 
 import wiro.annotation._
 
-package GameController {
-  @path("rps")
-  trait GameApi {
-    @query
-    def result(): Future[Either[NoGameInMemory, GameSummary]]
-
-    @command
-    def play(
-        userMove: Move
-    ): Future[Either[Throwable, Unit]]
-  }
-
-  class GameApiImpl(implicit ec: ExecutionContext) extends GameApi {
-    override def play(userMove: Move): Future[Either[Throwable, Unit]] =
-      Future {
-        Right(GameService.playGame(userMove))
-      }
+@path("rps")
+trait GameApi {
+  @query
+  def result(): Future[Either[NoGameInMemory, GameSummary]]
+
+  @command
+  def play(
+      userMove: Move
+  ): Future[Either[Throwable, Unit]]
+}
+
+class GameApiImpl(implicit ec: ExecutionContext, repo: InMemoryGameRepository) extends GameApi {
+  private val service = GameService()
```

Comment:
you shouldn't instantiate the service here. You should receive it as an explicit dependency similarly to what you do with `repo` above (just, explicit for readability in the Boot).

The idea is that `GameApiImpl` doesn't know about the concrete implementation of `GameService`, it just knows its api. Read the other comments for more context.